processor/resourcedetectionprocessor/internal/gcp/gcp.go

```diff
@@ -52,6 +53,18 @@ type detector struct {
 }
 
 func (d *detector) Detect(context.Context) (resource pcommon.Resource, schemaURL string, err error) {
+	if d.detector.CloudPlatform() == gcp.BareMetalSolution {
+		d.rb.SetCloudProvider(conventions.AttributeCloudProviderGCP)
+		errs := d.rb.SetFromCallable(d.rb.SetCloudAccountID, d.detector.BareMetalSolutionProjectID)
+
+		d.rb.SetCloudPlatform("gcp_bare_metal_solution")
```

Comment:
We should add this to the semantic conventions to make this an official/stable value.  E.g. https://github.com/open-telemetry/semantic-conventions/blob/7c3b66874ceae69e9489605c48c92e81b9b600ca/model/registry/cloud.yaml#L182

Reply:
Thanks for bringing this up! I've actually already addressed this in [this pull request](https://github.com/open-telemetry/semantic-conventions/pull/64). The reason I'm still using the string literal "gcp_bare_metal_solution" here is because the current version of the semconv module used by the opentelemetry-collector-contrib hasn't yet incorporated these changes. I'll update this once the new version of semconv is integrated into the collector.

Reply:
perfect